rotkehlchen/data_import/importers/rotki_events.py

```diff
@@ -0,0 +1,122 @@
+    deserialize_asset_amount,
+    deserialize_timestamp_from_date,
+)
+from rotkehlchen.types import Fee, Location, TimestampMS
+
+GENERIC_TYPE_TO_HISTORY_EVENT_TYPE_MAPPINGS = {
+    'Deposit': (HistoryEventType.DEPOSIT, HistoryEventSubType.SPEND),
+    'Withdrawal': (HistoryEventType.WITHDRAWAL, HistoryEventSubType.RECEIVE),
+    'Income': (HistoryEventType.RECEIVE, HistoryEventSubType.RECEIVE),
+    'Loss': (HistoryEventType.SPEND, HistoryEventSubType.NONE),
+    'Staking': (HistoryEventType.STAKING, HistoryEventSubType.REWARD),
+}
+
+
+class RotkiGenericEventsImporter(BaseExchangeImporter):
+    def _consume_rotki_events(
+            self,
+            cursor: DBCursor,
+            csv_row: Dict[str, Any],
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S',
+    ) -> None:
+        """Consume Rotki generic events import CSV file.
+        May raise:
+        - UnsupportedCSVEntry if an unknown type is encountered.
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        """
+        try:
+            event_type, event_subtype = GENERIC_TYPE_TO_HISTORY_EVENT_TYPE_MAPPINGS[csv_row['Type']]  # noqa: E501
+        except KeyError as e:
+            raise UnsupportedCSVEntry(f'Unsupported entry {csv_row["Type"]}. Data: {csv_row}') from e  # noqa: E501
+        events: List[HistoryBaseEntry] = []
+        identifier = csv_row['ID']
+        timestamp = TimestampMS(
+            deserialize_timestamp_from_date(
+                date=csv_row['Timestamp'],
+                formatstr=timestamp_format,
+                location='Rotki generic events importer',
+            ) * 1000,
+        )
+        fee = Fee(deserialize_asset_amount(csv_row['Fee'])) if csv_row['Fee'] else Fee(ZERO)  # noqa: E501
+        fee_currency = (
+            symbol_to_asset_or_token(csv_row['Fee Currency'])
+            if csv_row['Fee Currency'] and fee is not None else None
+        )
+        history_event = HistoryBaseEntry(
+            event_identifier=HistoryBaseEntry.deserialize_event_identifier_from_kraken(identifier),
+            sequence_index=csv_row['sequence_index'],
+            timestamp=timestamp,
+            location=Location.EXTERNAL,
+            event_type=event_type,
+            event_subtype=event_subtype,
+            asset=symbol_to_asset_or_token(csv_row['Currency']),
+            balance=Balance(
+                amount=deserialize_asset_amount(csv_row['Amount']),
+                usd_value=ZERO,
+            ),
+            notes=csv_row['Description'],
+        )
+        events.append(history_event)
+        if fee != ZERO and fee_currency is not None:
+            fee_event = HistoryBaseEntry(
+                event_identifier=HistoryBaseEntry.deserialize_event_identifier_from_kraken(identifier),  # noqa: E501
+                sequence_index=csv_row['sequence_index'] + 1,
+                timestamp=timestamp,
+                location=Location.EXTERNAL,
```

Comment:
Would it be interesting for the user to customize also the location? For example if I want to add some profits in a product of `location X` that we don't track?